2.0/Dockerfile

```diff
@@ -33,11 +33,14 @@ RUN set -ex \
 		/usr/share/elasticsearch/config/scripts \
 	; do \
 		mkdir -p "$path"; \
-		chown -R elasticsearch:elasticsearch "$path"; \
 	done
 
 COPY config/logging.yml /usr/share/elasticsearch/config/
 
+RUN sed -i -E "s/# network.host:.*/network.host: 0.0.0.0/" /etc/elasticsearch/elasticsearch.yml \
+	&& cp /etc/elasticsearch/elasticsearch.yml /usr/share/elasticsearch/config \
+	&& chown -R elasticsearch:elasticsearch /usr/share/elasticsearch/
```

Comment:
I'm not sure I understand why we're bothering with `/etc/elasticsearch/elasticsearch.yml` here?
```console
$ docker run -it --rm elasticsearch:2 bash
root@b73df41d8044:/# grep -v '^#' /etc/elasticsearch/elasticsearch.yml
root@b73df41d8044:/# 
```

Also, we need to somehow reconcile with the `Dockerfile.template` at the top-level, which I'm thinking the easiest way is probably to take advantage of the fact that our `config` subdirectories are independent, ie:
```diff
diff --git a/2.0/config/elasticsearch.yml b/2.0/config/elasticsearch.yml
new file mode 100644
index 0000000..5108c77
--- /dev/null
+++ b/2.0/config/elasticsearch.yml
@@ -0,0 +1 @@
+network.host: 0.0.0.0
diff --git a/Dockerfile.template b/Dockerfile.template
index cb96d3e..9ce7d5a 100644
--- a/Dockerfile.template
+++ b/Dockerfile.template
@@ -36,7 +36,7 @@ RUN set -ex \
 		chown -R elasticsearch:elasticsearch "$path"; \
 	done
 
-COPY config/logging.yml /usr/share/elasticsearch/config/
+COPY config /usr/share/elasticsearch/config
 
 VOLUME /usr/share/elasticsearch/data
 
```